cmake/AddInputFileTests.cmake

```diff
@@ -1,7 +1,8 @@
 # Distributed under the MIT License.
 # See LICENSE.txt for details.
 
-find_package(PythonInterp REQUIRED)
+include(SpectreFindPython)
+spectre_find_python(REQUIRED COMPONENTS Interpreter)
```

Comment:
I intentionally don't want it invoked inside `FindNumPy`... We could do `find_package(SpectrePython...)`, but I'm not sure that gains us anything

Reply:
Do you have an objection to calling it `FindSpectrePython` and invoking it with `find_package`? That seems the cleanest solution to me. It means we could replace `find_package(SpectrePython)` -> `find_package(Python)` once we move to cmake 3.12

Reply:
I will see how easy it is to do, but first want to get your feedback on the following :) We won't be able to replace `find_package(SpectrePython...)` since `SpectrePython` would also support `SciPy`, which `Python` doesn't. Now, with newer SciPy (I don't know which version) one doesn't need to find SciPy include directories. If we can restrict to that it would be nice. However, right now we can also `spectre_find_python(` -> `find_package(Python`, which is no more difficult of a find-replace... I guess as I'm writing this I'm coming to the conclusion that switching to `find_package(SpectrePython` might be a decent amount of work with minimal to no benefit :)

Reply:
Here's another idea: can we just copy https://github.com/Kitware/CMake/blob/master/Modules/FindPython.cmake to external?

Reply:
I'll see if that works or if it causes conflicts. It might also cause lots of warnings when turning on `--warn-uninitialized`. I'll give it a shot and report back :)

Reply:
I looked into copying that over and it's not really great... The cleanest I think is to stick with what we have here